src/PrimordialChem/CMakeLists.txt

```diff
@@ -1,5 +1,10 @@
-add_executable(test_primordial_chem test_primordial_chem.cpp ../main.cpp ../CloudyCooling.cpp ../GrackleDataReader.cpp ../Chemistry.cpp ${primordial_chem_sources})
-target_compile_definitions(test_primordial_chem PUBLIC PRIMORDIAL_CHEM) #this will add #define PRIMORDIAL_CHEM
+set(microphysics_network_name "primordial_chem") #this will override network_name to primordial_chem for this directory only
+setup_target_for_microphysics_compilation(${microphysics_network_name} "${CMAKE_CURRENT_BINARY_DIR}/")
+
+add_executable(test_primordial_chem test_primordial_chem.cpp ../main.cpp ../GrackleDataReader.cpp ../Chemistry.cpp ${primordial_chem_sources})
+
+include_directories(${primordial_chem_dirs} "${CMAKE_CURRENT_BINARY_DIR}/" "includes/extern_parameters.H" "includes/network_properties.H")
+target_compile_definitions(test_primordial_chem PUBLIC CHEM) #this will add #define CHEM
 
```

Comment:
removed CloudyCooling.cpp from primordial chem test, because of the non-constant gamma issue